packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -212,6 +212,27 @@ class IOSDevice extends Device {
 
   DevicePortForwarder _portForwarder;
 
+  /// Uses `nslookup` to get the IP address of this device.
+  Future<String> getIP(bool ipv6) async {
```

Comment:
This logic doesn't seem complete, or robust enough for process exceptions or unexpected output.  For example, this is what I get when I run it (particularly re: "last match because 1st one is IPv6"):
```
$ nslookup iPhone                                     
Server:		192.168.0.1
Address:	192.168.0.1#53

Non-authoritative answer:
Name:	iPhone
Address: 23.202.231.169
Name:	iPhone
Address: 23.217.138.110
```

Reply:
Would it be too hard to just call `getaddrinfo` directly instead of using `nslookup`? I'm considering doing this, so I don't have to parse the output of this tool.

Reply:
The tool doesn't use any `ffi`.